### ISSUE TITLE
Dependencies are not resolved

### ISSUE THREAD

**Issue by vitalii-dmytruk** [morewings/cra-template-npm-library#723]
Creating a library using the template cause the nested error.

```

Installing template dependencies using npm...
npm WARN idealTree Removing dependencies.react-scripts in favor of devDependencies.react-scripts
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: testlib@0.1.0
npm ERR! Found: react@18.2.0
npm ERR! node_modules/react
npm ERR!   react@"^18.2.0" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
```

**Comment by morewings** [comment]
@vitalii-dmytruk is this one time error or does it persist?

**Comment by vitalii-dmytruk** [comment]
It persists.
Sorry, I didn't realize I'd posted an incomplete log. Here is the full one

```
/usr/local/bin/npm install
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: tslib@0.1.0
npm ERR! Found: react@18.2.0
npm ERR! node_modules/react
npm ERR!   react@"^18.2.0" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer react@"^16.9.0 || ^17.0.0" from @testing-library/react-hooks@8.0.1
npm ERR! node_modules/@testing-library/react-hooks
npm ERR!   dev @testing-library/react-hooks@"8.0.1" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! See /Users/salo/.npm/eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/salo/.npm/_logs/2022-12-12T10_13_09_147Z-debug-0.log

Process finished with exit code 1
```

**Comment by morewings** [comment]
@vitalii-dmytruk fixed in `1.2.2`. Thanks for reporting
